### PR TITLE
[Docs] Add source links and API reference to visual-vocabulary README

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
- Adding two additional columns to the README
- Rename heatmap matrix to correlation matrix

After 2 approvals I would rename the entire folder _visual-vocabulary to visual-vocabulary given that it's public now 👍 

## Screenshot
Screenshot 2024-09-19 at 11 07 02

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/examples/_visual-vocabulary/README.md [hunk 1]
```diff
@@ -29,83 +29,85 @@ Inspired by:
 Credits and sources:
 
 - Charting library: [Plotly](https://plotly.com/python/plotly-express/)
-
+- Plotly Express chart examples: [Plotly Express](https://plotly.com/python/plotly-express/)
```
Comment (petar-qb) [inline vizro-core/examples/_visual-vocabulary/README.md:32]: Why do we link (https://plotly.com/python/plotly-express/) the same thing twice, one below the other?

**Review by petar-qb — APPROVED** [review]
Looks good to me.

File: vizro-core/examples/_visual-vocabulary/README.md (hunk 1 above)
Comment (huong-li-nguyen) [reply vizro-core/examples/_visual-vocabulary/README.md:32]: Ups! The first one should be the git repo - I'll change that

**Review by nadijagraca — APPROVED** [review]
Looks good.

**Pull request merged**